lib/AppInfo/Application.php

```diff
@@ -174,7 +174,7 @@ public function __construct(array $urlParams = [])
 
         $container->registerService(Favicon::class, function (IContainer $c): Favicon {
             $favicon = new Favicon();
-            $settings = ['dir' => sys_get_temp_dir()];
+            $settings = ['dir' => \OC::$server->getTempManager()->getTempBaseDir()];
```

Comment:
You may fetch the temp manager from container but not sure if this works here.

```
			/** @var ITempManager $tempManager */
			$tempManager = $c->query(ITempManager::class);
			$favicon = new Favicon();
			$settings = ['dir' => $tempManager->getTempBaseDir()];
			$favicon->cache($settings);
			return $favicon;
```

(dont forget to import ITempManager if you use the approach above)

Reply:
works and looks more consistent